pkg/kvstore/etcd.go

```diff
@@ -101,12 +108,60 @@ func NewEtcdClient(config *client.Config, cfgPath string) (KVClient, error) {
 				ec.session = newSession
 				ec.sessionMU.Unlock()
 				log.Debugf("Renewing etcd session")
+				if err := ec.CheckMinVersion(10 * time.Second); err != nil {
+					log.Fatalf("%s", err)
```

Comment:
To confirm, the logic here is:
* Renew the etcd session
* Check if the version does not meet our requirements

We do this in an infinite loop because the etcd version could change while Cilium is running, so we need to consistently keep checking?